### PR TITLE
ROX-10073: Postgres directly scoped SAC for Delete/DeleteMany method

### PR DESCRIPTION
**Pull request by janisz**
## Description

SAC filter for Delete and DeleteMany for directly scoped resources.

## Testing Performed

CI

### CODE REVIEW

File: tools/generate-helpers/pg-table-bindings/store.go.tpl [hunk 1]
```diff
@@ -711,30 +740,40 @@ func (s *storeImpl) GetMany(ctx context.Context, ids []{{$singlePK.Type}}) ([]*{
...
-        return sac.ErrResourceAccessDenied
     }
-    {{- else if .Obj.IsGloballyScoped }}
+{{- else if .Obj.IsGloballyScoped }}
     {{ template "defineScopeChecker" "READ_WRITE" }}
     if ok, err := scopeChecker.Allowed(ctx); err != nil {
         return err
     } else if !ok {
-        return sac.ErrResourceAccessDenied
+        return nil
```
Comment (rhybrillou) [inline tools/generate-helpers/pg-table-bindings/store.go.tpl:753]: Why this change in error management ?

**Review by rhybrillou — COMMENTED** [review]
Looks good, apart from a small glitch on the called PermissionChecker function called in DeleteMany

File: pkg/search/postgres/common.go [hunk 2]
```diff
@@ -519,3 +519,19 @@ func RunCountRequestForSchema(schema *walker.Schema, q *v1.Query, db *pgxpool.Po
...
+	if err != nil || query == nil {
+		return err
+	}
+
+	queryStr := query.String()
+	rows, err := db.Query(context.Background(), replaceVars(queryStr), query.Data...)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
```
[the remark this answers is not in the document]
Comment (janisz) [reply pkg/search/postgres/common.go:535]: Good point! Actually we should not use `Query` here but `Exec`
